www/js/control/ControlSyncHelper.tsx

```diff
@@ -0,0 +1,284 @@
+        entry_data = await getTransition(getEndTransitionKey());
+        messagePut = await window.cordova.plugins.BEMUserCache.putMessage(sensorKey, entry_data);
+        forceSync();
+    };
+
+    return (
+        <>
+            <SettingRow textKey="control.force-sync" iconName="sync" action={forceSync}></SettingRow>
+            <SettingRow textKey="control.end-trip-sync" iconName="sync-alert" action={endForceSync}></SettingRow>
+
+            {/* dataPending */}
+            <Modal visible={dataPendingVis} onDismiss={()=>setDataPendingVis(false)} transparent={true}>
+                <Dialog visible={dataPendingVis} 
+                        onDismiss={()=>setDataPendingVis(false)} 
+                        style={settingStyles.dialog(colors.elevation.level3)}>
+                    <Dialog.Title>{t('data pending for push')}</Dialog.Title>
+                    <Dialog.Actions>
+                        <Button onPress={()=>{
+                            setDataPendingVis(false);
+                            Logger.log("user refused to re-sync")}}>
+                                {t('general-settings.cancel')}
+                        </Button>
+                        <Button onPress={()=>{
+                            setDataPendingVis(false);
+                            forceSync();}}>
+                                {t('general-settings.confirm')}
+                        </Button>
+                    </Dialog.Actions>
+                </Dialog>
+            </Modal>
+
+            <AlertBar visible={dataPushedVis} setVisible={setDataPushedVis} messageKey='all data pushed!'></AlertBar>
+        </>
+    )
+}
+
+//UI for editing the sync config
+const ControlSyncHelper = ({ editVis, setEditVis }) => {
+    const { t } = useTranslation(); 
+    const { colors } = useTheme();
+    const CommHelper = getAngularService("CommHelper");
+    const Logger = getAngularService("Logger");
+
+    const [ localConfig, setLocalConfig ] = useState<syncConfig>();
+    const [ intervalVis, setIntervalVis ] = useState<boolean>(false);
+
+    /* 
+     * Functions to read and format values for display
+     */
+    async function getSyncSettings() {
+        let tempConfig = await getConfig();
+        setLocalConfig(tempConfig);
+    }
+
+    useEffect(() => {
+        getSyncSettings();
+    }, [editVis])
+
+    const syncIntervalActions = [
+        {text: "1 min", value: 60},
+        {text: "10 min", value: 10 * 60},
+        {text: "30 min", value: 30 * 60},
+        {text: "1 hr", value: 60 * 60}
+    ]
+
+    /* 
+     * Functions to edit and save values
+     */
+    async function saveAndReload() {
+        console.log("new config = "+localConfig);
+        try{
+            let set = setConfig(localConfig);
+            //NOTE -- we need to make sure we update these settings in ProfileSettings :) -- getting rid of broadcast handling for migration!!
```

Comment:
Because the ControlSyncHelper contains the table and editor, the updates immediately show in the UI on save, removing this comment to prevent further confusion.